tests/memory_checker/test_memory_checker.py

```diff
@@ -420,238 +373,289 @@ def get_container_mem_usage(duthost, container_name):
+
+    @property
+    def memory_service_name(self):
+        return 'container_memory_{}'.format(self.name)
+
+    def is_running(self):
+        return is_container_running(self.duthost, self.name)
+
+    def get_monit_mem_status(self):
+        return get_monit_service_status(self.duthost, self.memory_service_name)
+
+    def is_monit_mem_ok(self):
+        status = self.get_monit_mem_status()
+        return status['status'] == 'Status ok'
+
+    def is_monit_mem_failed(self):
+        status = self.get_monit_mem_status()
+        return status['status'] == 'Status failed'
+
+    def is_monit_mem_last_ok(self):
+        status = self.get_monit_mem_status()
+        return status['status'] == 'Status ok' and status['last_exit_value'] == '0'
+
+    def is_monit_mem_last_failed(self):
+        status = self.get_monit_mem_status()
+        return status['status'] == 'Status ok' and status['last_exit_value'] != '0'
+
+    def remove(self):
+        remove_container(self.duthost, self.name)
+
+    def restart(self):
+        restart_container(self.duthost, self.name)
+
+    def post_check(self):
+        postcheck_critical_processes(self.duthost, self.name)
+
+    def start_consume_memory(self):
+        start_consume_memory(self.duthost, self)
+
+    def stop_consume_memory(self):
+        stop_consume_memory(self.duthost, self)
+
+    def get_restart_expected_logre(self):
+        cap_name = self.name.capitalize()
+        return [
+            r".*restart_service.*Restarting service '{}'.*".format(self.name),
+            r".*Stopping {} container.*".format(cap_name),
+            r".*Stopped {} container.*".format(cap_name),
+            r".*Starting {} container.*".format(cap_name),
+            r".*Started {} container.*".format(cap_name),
+        ]
+
+    def get_last_start_date(self):
+        rc = self.duthost.shell(r"docker inspect --format \{{\{{.State.StartedAt\}}\}} {}".format(self.name))
+        date_str = rc['stdout_lines'][0].strip()
+        return dateutil.parser.isoparse(date_str)
+
+    def has_container_restarted(self):
+        start_date = self.get_last_start_date()
+        return start_date > self._last_start_date
+
+    def wait_restarted(self, start_date=None):
+        self._last_start_date = start_date or self.get_last_start_date()
+        restarted = wait_until(CONTAINER_RESTART_THRESHOLD_SECS, 1, 0,
+                               self.has_container_restarted)
+        pytest_assert(restarted, "Failed to restart '{}' container!".format(self.name))
+        self._last_start_date = None
+
+    def wait_monit_mem_ok(self, timeout=MONIT_MEMORY_CHECK_TIMEOUT):
+        logger.info("Waiting for monit status ok for %s", self.name)
+        res = wait_until(timeout, 1, 0, self.is_monit_mem_ok)
+        pytest_assert(res, "Failed to wait for one monit cycle to be ok for {}".format(self.name))
+
+    def wait_monit_mem_failed(self, timeout=MONIT_MEMORY_CHECK_TIMEOUT):
```

Comment:
default timeout for this function is `MONIT_MEMORY_CHECK_TIMEOUT = 700`, but caller of it uses 10sec and 60sec, why such difference?

Reply:
`MONIT_MEMORY_CHECK_TIMEOUT` is set considering the default monit parameters.
So 60s cycles and 10 cycles required for the telemetry memory check to fail so that 600s with an added 60s in case the memory condition started right after a monit cycle and some extra 40s.

To avoid the tests from taking ages, they are parametrized with different monit cycle times.
Regarding the 2 usecases:
 - 10s -> essentially we expect the status to be failed on the first call if it's not the test should be failed
 - 60s -> 6x the time it should take for the 10 cycles of 1s to mark the status as failed (I explained this more in your other comment)